unittest/syncd/TestVendorSai.cpp

```diff
@@ -37,6 +43,96 @@ static sai_service_method_table_t test_services = {
     profile_get_next_value
 };
 
+class VendorSaiTest : public ::testing::Test
+{
+public:
+    VendorSaiTest() = default;
+    virtual ~VendorSaiTest() = default;
+
+public:
+    virtual void SetUp() override
+    {
+        m_vsai = std::make_shared<VendorSai>();
```

Comment:
this is not required in setup, this init and teardown should be in the test itself

Reply:
@kcudnik having the same instruction in each test will lead to code duplication. The whole idea of using this fixture is to minimize amount of the identical code and let the user to focus on API testing. Also, here we want to force a cleanup when the test fails unexpectedly. Please elaborate on why it is a wrong approach?